net: shell: conn_mgr shell compilation error

**Description of bug**
Compilation error when using Net Shell

**To Reproduce**

**Expected behavior**
No compilation warnings

**Impact**
Compilation error, appears to still function correctly

**Logs and console output**

> [268/437] Building C object zephyr/subsys/net/CMakeFiles/subsys__net.dir/lib/shell/cm.c.obj
> ../zephyrNew/subsys/net/lib/shell/cm.c: In function 'cm_get_iface_info':
> ../zephyrNew/subsys/net/lib/shell/cm.c:371:2: warning: implicit declaration of function 'snprintf' [-Wimplicit-function-declaration]
>   371 |  snprintf(buf, len, "%d (%p - %s - %s)", net_if_get_by_iface(iface), iface, name,
>       |  ^~~~~~~~
> ../zephyrNew/subsys/net/lib/shell/cm.c:371:2: warning: incompatible implicit declaration of built-in function 'snprintf'
> ../zephyrNew/subsys/net/lib/shell/cm.c:21:1: note: include '<stdio.h>' or provide a declaration of 'snprintf'
>    20 | #include <zephyr/net/conn_mgr_monitor.h>
>   +++ |+#include <stdio.h>

**Environment (please complete the following information):**

-- CMake version: 3.25.0-rc2
-- Found Python3: C:\Python311\python.exe (found suitable version "3.11.9", minimum required is "3.8") found components: Interpreter
-- Zephyr version: 3.7.99 (/zephyrNew)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: particle_argon, qualifiers: nrf52840
-- Found toolchain: gnuarmemb (C:\gnu_arm_embedded_10)
-- Found Dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.5.0", minimum required is "1.4.6")

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->

--- a/subsys/net/lib/shell/cm.c
+++ b/subsys/net/lib/shell/cm.c
@@ -368,10 +368,10 @@ static void cm_get_iface_info(struct net_if *iface, char *buf, size_t len)
 		strcpy(name, CM_IF_NAME_NONE);
 	}
 
-	snprintf(buf, len, "%d (%p - %s - %s)", net_if_get_by_iface(iface), iface, name,
+	snprintk(buf, len, "%d (%p - %s - %s)", net_if_get_by_iface(iface), iface, name,
 						iface2str(iface, NULL));
 #else
-	snprintf(buf, len, "%d (%p - %s)", net_if_get_by_iface(iface), iface,
+	snprintk(buf, len, "%d (%p - %s)", net_if_get_by_iface(iface), iface,
 					   iface2str(iface, NULL));
 #endif
 }

